java/lang/String/String.cpp

```diff
@@ -845,4 +850,170 @@ boolean String::operator>=(const String &target) const {
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, long value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned short value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned int value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned long value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, double value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, float value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, char *value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, Short value)
+{
+    return String::print(format, value.shortValue());
+}
+
+String String::print(const String &format, Integer value)
+{
+    return String::print(format, value.intValue());
+}
+
+String String::print(const String &format, Long value)
+{
+    return String::print(format, value.longValue());
+}
+
+String String::print(const String &format, Float value)
+{
+    return String::print(format, value.floatValue());
+}
+
+String String::print(const String &format, Double value)
+{
+    return String::print(format, value.doubleValue());
+}
+
+String String::print(const String &format, String value)
+{
+    return String::print(format, value.toString());
+}
+
+String String::format(const String &format)
+{

```

Comment:
{ should almost always be at the end of the previous line. We follow the google code style for C++.